### PR TITLE
gs-details-page: Multiple UI fixes

### PR DESCRIPTION
**Pull request by uajain**
https://phabricator.endlessm.com/T27332

### CODE REVIEW

**Comment by wjt** [comment]
(Why the `trivial: ` prefix? Why are these 3 changes all in 1 commit?)

File: src/gs-details-page.ui [hunk 1]
```diff
@@ -153,15 +153,18 @@
...
                             <property name="spacing">9</property>
                             <child>
-                              <object class="GtkLinkButton" id="button_details_add_shortcut">
+                              <object class="GtkButton" id="button_details_add_shortcut">
                                 <property name="visible">False</property>
                                 <property name="can_focus">True</property>
                                 <property name="use_underline">True</property>
                                 <property name="label" translatable="yes" comments="Translators: A label for a button to add a shortcut to the selected application.">_Add to Desktop</property>
+                                <style>
+                                  <class name="suggested-action"/>
```
Comment (pwithnall) [inline src/gs-details-page.ui]: Does this need to change dynamically as different buttons are made visible/hidden?

File: src/gs-details-page.c [hunk 2]
```diff
@@ -1877,6 +1877,9 @@ _set_app (GsDetailsPage *self, GsApp *app)
 	g_signal_connect_object (self->app, "notify::license",
 				 G_CALLBACK (gs_details_page_notify_state_changed_cb),
 				 self, 0);
+	g_signal_connect_object (self->app, "notify::quirk",
+				 G_CALLBACK (gs_details_page_notify_state_changed_cb),
+				 self, 0);
```
Comment (pwithnall) [inline src/gs-details-page.c]: This should definitely go upstream.

**Comment by pwithnall** [comment]
I agree with all of Will's suggestions too.

File: src/gs-details-page.ui (hunk 1 above)
Comment (uajain) [reply src/gs-details-page.ui]: Not sure why being dynamic here? I have taken this change from `eos3.6` branch (i.e. this is a fallout from the 3.32 rebase.) 

Both `_add_shortcut` and `_remove_shortcut` are separate buttons and visibility is toggled according to the given situation in the icon-grid. Not sure why we need to make the style class being dynamically applied

Comment (pwithnall) [reply src/gs-details-page.ui]: As in, buttons on the details page are often shown/hidden based on the state of the app. We don’t ever want the situation where *two* buttons are visible, both with class `suggested-action`.

I don’t know if that’s the case after applying this PR, but please check.

**Comment by uajain** [comment]
> I believe we are the only user of this (upstream) UI. It was originally added upstream by Joaquim. Shouldn't most/all of this PR go upstream too?

Sorry, I was not aware aware of this fact! I never saw desktop shortcuts in gnome-software hence, assumed its downstream only. I will push all this upstream

File: src/gs-details-page.ui (hunk 1 above)
Comment (uajain) [reply src/gs-details-page.ui]: No, that's not the case. I can see it in action. It is making the behavior consistent with what we have/had in `3.6` OS's gnome-software. 

(I have a 3.6 release OS image VM which I use to compare and contrast these UI fixes).

**Comment by uajain** [comment]
I have upstreamed most of the changes : https://gitlab.gnome.org/GNOME/gnome-software/merge_requests/295

Just one changed ineeds to kept downstream, which is the "Uninstall" button / UI placement.

File: src/gs-details-page.ui (hunk 1 above)
Comment (wjt) [reply src/gs-details-page.ui]: I think Philip's point is that both this button, and "Launch", have the `suggested-action` style class. Won't this mean that, when the app is installed but not on the desktop, both "Launch" and "Add to Desktop" will be blue? Could you take screenshots of the 3 states (not installed, installed but not on desktop, installed and on desktop)?

Comment (uajain) [reply src/gs-details-page.ui]: **Not installed**
Screenshot from 2019-08-01 20-08-39


**Installed and not on desktop** 
Screenshot from 2019-08-01 20-08-15


**Installed and  on desktop**
Screenshot from 2019-08-01 20-08-22

Comment (wjt) [reply src/gs-details-page.ui]: https://developer.gnome.org/hig/stable/buttons.html.en

> Each view should only include a single suggested or destructive button.

De-emphasising Launch was a deliberate downstream change, see discussion in https://phabricator.endlessm.com/T14949 and designs linked there. I think we should just drop the `suggested-action` on the launch button.

Comment (uajain) [reply src/gs-details-page.ui]: P

> https://developer.gnome.org/hig/stable/buttons.html.en
> 
> > Each view should only include a single suggested or destructive button.
> 
> De-emphasising Launch was a deliberate downstream change, see discussion in https://phabricator.endlessm.com/T14949 and designs linked there. I think we should just drop the `suggested-action` on the launch button.

Pushed a `fixup!` which is linking where this particular changed should have been introduced downstream. Also, made a note in the commit message of the original commit_id on `eos3.6` branch.

**Pull request merged**